### PR TITLE
Remove implicit any's

### PR DESCRIPTION
**Pull request by edgarmueller**
This is a very big PR split up into multiple commits which deal with removing the implicit any warnings (see #1072 )

### CODE REVIEW

File: packages/material-tree-renderer/src/tree/ObjectListItem.tsx [hunk 1]
```diff
@@ -4,12 +4,12 @@ import { indexFromPath } from '../helpers/util';
...
 import { connect } from 'react-redux';
 import {
-  getData,
-  getSchema,
-  JsonSchema7,
-  Paths,
-  resolveData,
-  update
+    getData,
+    getSchema, JsonFormsState,
```
Comment (eneufeld) [inline packages/material-tree-renderer/src/tree/ObjectListItem.tsx]: formatting

File: packages/material-tree-renderer/src/tree/TreeWithDetailRenderer.tsx [hunk 2]
```diff
@@ -4,34 +4,31 @@ import * as _ from 'lodash';
...
     findUISchema,
     getData,
-    JsonSchema7,
+    JsonFormsState,
+    JsonSchema,
+    OwnPropsOfControl,
     Paths,
     Resolve,
-    Runtime,
+    Runtime, StatePropsOfControl,
```
Comment (eneufeld) [inline packages/material-tree-renderer/src/tree/TreeWithDetailRenderer.tsx]: formatting

File: packages/material/src/controls/MaterialDateControl.tsx [hunk 3]
```diff
@@ -25,121 +25,132 @@
...
-  mapStateToControlProps,
-  RankedTester,
-  rankWith,
-  StatePropsOfControl
+    computeLabel,
+    ControlProps,
+    ControlState,
+    isDateControl,
+    isDescriptionHidden,
+    isPlainLabel, JsonFormsState,
```
Comment (eneufeld) [inline packages/material/src/controls/MaterialDateControl.tsx]: formatting

File: packages/vanilla/src/controls/InputControl.tsx [hunk 4]
```diff
@@ -24,23 +24,24 @@
...
-  ControlState,
-  formatErrorMessage,
-  isControl,
-  isDescriptionHidden,
-  isPlainLabel,
-  mapStateToControlProps,
-  NOT_APPLICABLE,
-  RankedTester,
-  rankWith,
+    computeLabel, ControlProps,
```
Comment (eneufeld) [inline packages/vanilla/src/controls/InputControl.tsx]: formatting

File: packages/vanilla/src/controls/RadioGroupControl.tsx [hunk 5]
```diff
@@ -25,7 +25,7 @@
 import * as React from 'react';
 import {
     computeLabel,
-    ControlElement,
+    ControlElement, ControlProps,
```
Comment (eneufeld) [inline packages/vanilla/src/controls/RadioGroupControl.tsx]: formatting

File: packages/vanilla/src/layouts/GroupLayout.tsx [hunk 6]
```diff
@@ -25,16 +25,16 @@
...
 import {
-  GroupLayout,
-  mapStateToLayoutProps,
-  RankedTester,
-  rankWith,
-  uiTypeIs,
+    GroupLayout,
+    mapStateToLayoutProps,
+    RankedTester,
+    rankWith, RendererProps,
```
Comment (eneufeld) [inline packages/vanilla/src/layouts/GroupLayout.tsx]: formatting

File: packages/vanilla/src/layouts/HorizontalLayout.tsx [hunk 7]
```diff
@@ -24,26 +24,25 @@
...
 import {
-  HorizontalLayout,
-  mapStateToLayoutProps,
-  RankedTester,
-  rankWith,
-  uiTypeIs,
+    HorizontalLayout,
+    mapStateToLayoutProps,
+    RankedTester,
+    rankWith, RendererProps,
```
Comment (eneufeld) [inline packages/vanilla/src/layouts/HorizontalLayout.tsx]: formatting

File: packages/vanilla/src/layouts/VerticalLayout.tsx [hunk 8]
```diff
@@ -24,17 +24,17 @@
...
 import * as React from 'react';
 import {
-  mapStateToLayoutProps,
-  RankedTester,
-  rankWith,
-  uiTypeIs,
-  VerticalLayout,
+    mapStateToLayoutProps,
+    RankedTester,
+    rankWith, RendererProps,
```
Comment (eneufeld) [inline packages/vanilla/src/layouts/VerticalLayout.tsx]: formatting

File: packages/webcomponent/src/JsonFormsElement.tsx [hunk 9]
```diff
@@ -27,13 +27,13 @@ import * as ReactDOM from 'react-dom';
...
-  getUiSchema,
-  JsonFormsState,
-  JsonFormsStore
+    Actions,
+    Generate,
+    getData,
+    getSchema,
+    getUiSchema,
+    JsonFormsState,
+    JsonFormsStore, JsonSchema, UISchemaElement
```
Comment (eneufeld) [inline packages/webcomponent/src/JsonFormsElement.tsx]: formatting

File: packages/core/src/actions/index.ts [hunk 10]
```diff
@@ -63,6 +72,12 @@ export const init = (
       ajv
     });
 
+export interface RegisterDefaultDataAction {
+    type: 'jsonforms/ADD_DEFAULT_DATA';
```
[the remark this answers is not in the document]
Comment (edgarmueller) [reply packages/core/src/actions/index.ts]: These are singleton types, the constants are not applicable here.

**Pull request merged**